files/SZWESAMP/ZWEIKRA1

```diff
@@ -0,0 +1,221 @@
+//*      * NONSTRICT_VERIFY_CERTIFICATES and define ROOTZFCA, set
+//*      * IFROZFCA to 1 to connect z/OSMF certificate authority to
+//*      * Zowe keyring. Otherwise set to 0.
+//         SET IFROZFCA=0
+//*******************************************************************
+//*
+//*      * You do NOT need to set USERNAME when running ZWEGENER.
+//*      * This is used to keep some lines under the column limit.
+//*
+//         SET USERNAME={zowe.setup.security.users.zowe}
+//*
+//*********************************************************************
+//*
+//* EXECUTE COMMANDS FOR SELECTED SECURITY PRODUCT
+//*
+//RUNACF2  EXEC PGM=IKJEFT01,REGION=0M
+//SYSTSPRT DD SYSOUT=*
+//SYSTSIN  DD DDNAME=ACF2
+//ACF2     DD DATA,DLM=$$,SYMBOLS=JCLONLY
+ACF
+//
+* Create the keyring .............................................. */
+  SET PROFILE(USER) DIVISION(KEYRING)
+  INSERT {zowe.setup.security.users.zowe}.ZOWERING +
+  RINGNAME({zowe.setup.certificate.keyring.name})
+  F ACF2,REBUILD(USR),CLASS(P),DIVISION(KEYRING)
+$$
+//RUNOPT1  EXEC PGM=IKJEFT01,REGION=0M
+//SYSTSPRT DD SYSOUT=*
+//SYSTSIN  DD DDNAME=ACF2
+//ACF2     DD DATA,DLM=$$,SYMBOLS=JCLONLY
+ACF
+//
+* Option 1 - Default Option - BEGINNING ........................... */
+* Create Zowe's local CA authority ................................ */
+  SET PROFILE(USER) DIVISION(CERTDATA)
+  GENCERT CERTAUTH.ZOWECA -
+          LABEL({zowe.setup.certificate.keyring.caLabel}) SIZE(2048) -
+          SUBJSDN(CN='{zowe.setup.certificate.dname.caCommonName}' -
+                  OU='{zowe.setup.certificate.dname.orgUnit}' -
+                  O='{zowe.setup.certificate.dname.org}' -
+                  L='{zowe.setup.certificate.dname.locality}' -
+                  SP='{zowe.setup.certificate.dname.state}' -
+                  C='{zowe.setup.certificate.dname.country}') -
+  EXPIRE(05/01/30) -
+  KEYUSAGE(CERTSIGN)
+*
+* Connect Zowe's local CA authority to the keyring ................ */
+  SET PROFILE(USER) DIVISION(CERTDATA)
+  CONNECT CERTDATA(CERTAUTH.ZOWECA) -
+  RINGNAME({zowe.setup.certificate.keyring.name}) -
+  KEYRING({zowe.setup.security.users.zowe}.ZOWERING) USAGE(CERTAUTH)
+  CHKCERT CERTAUTH.ZOWECA
+*
+* Create a certificate signed by local zowe's CA .................. */
+   SET PROFILE(USER) DIV(CERTDATA)
+   GENCERT {zowe.setup.security.users.zowe}.ZOWECERT -
+            SUBJSDN(CN='{zowe.setup.certificate.dname.commonName}' -
+                    OU='{zowe.setup.certificate.dname.orgUnit}.' -
+                    O='{zowe.setup.certificate.dname.org}' -
+                    L='{zowe.setup.certificate.dname.locality}' -
+                    SP='{zowe.setup.certificate.dname.state}' -
+                    C='{zowe.setup.certificate.dname.country}') -
+           SIZE(2048) -
+           EXPIRE(05/01/30) -
+           LABEL({zowe.setup.certificate.keyring.label}) -
+           KEYUSAGE(HANDSHAKE) -
+           ALTNAME(IP=&IPADDRES DOMAIN={zowe.externalDomains.0}) -
```

Comment:
How do I reference entire array with unknown number of items?
This specific line is using an item with index 0.
But what if I don't know the number of items and want them all?

## Example
The number of lines can be setup individually by each user.
### Yaml
```
zowe:
  environments:
    jclHeaderLines:
      - "//ZWEJOB JOB 1984,"
      - "//   'Jonh Connor',NOTIFY=SARAH,MSGLEVEL=(T,800)"
```
### Template
```
{zowe.environments.jclHeaderLines}
//*
//* This program and the accompanying materials are made available
```
### Result
```
CFG.ZOWE.ENVIRONMENTS.JCLHEADERLINES                                
//*                                                                 
//* This program and the accompanying materials are made available  
```

Reply:
I guess enhancing the rexx code will give us a way to do this, but I've never ever seen a user with more than one entry for externalDomains.
So yes this could be better in theory but in practice there is no difference.